cli/waiter/subcommands/maintenance.py

```diff
@@ -0,0 +1,80 @@
+import requests
+
+from waiter import terminal, http_util
+from waiter.token_post import process_post_result, post_failed_message
+from waiter.querying import get_token, query_token
+from waiter.util import guard_no_cluster, logging, print_info
+
+
+def get_cluster_with_token(clusters, token_name):
+    query_result = query_token(clusters, token_name)
+    if query_result["count"] == 0:
+        raise Exception('The token does not exist. You must create it first.')
+    else:
+        cluster_names_with_token = list(query_result['clusters'].keys())
+        return next(c for c in clusters if c['name'] == cluster_names_with_token[0])
+
+
+def maintenance(clusters, args, _):
+    guard_no_cluster(clusters)
+    logging.debug('args: %s' % args)
+    token_name = args['token']
+    sub_action = args['sub_action']
+    cluster = get_cluster_with_token(clusters, token_name)
+    cluster_name = cluster['name']
+    cluster_url = cluster['url']
+    existing_token_data, existing_token_etag = get_token(cluster, token_name)
+    json_body = existing_token_data
+    headers = {'If-Match': existing_token_etag}
+    params = {'token': token_name}
+    try:
+        if sub_action == 'start':
+            update_doc = {"maintenance": {"message": args['message']}}
+            json_body.update(update_doc)
+        elif sub_action == 'stop':
+            if 'maintenance' in existing_token_data:
+                json_body.pop("maintenance")
+            else:
+                raise Exception("Token is not in maintenance mode")
```

Comment:
Is there a way to check if a token is in maintenance mode?
If not, should we add a `check` sub-command that exits non-zero if token is not in maintenance mode?